### PR TITLE
Added state control for updatable objects. 

### PR DESCRIPTION
**Pull request by vitenti**
Added testing and fixed pylint issues (old and new) on tests (including a pylintrc for tests and its inclusion on GHA).

### CODE REVIEW

File: firecrown/parameters.py [hunk 1]
```diff
@@ -63,16 +63,16 @@ class RequiredParameters:
     which implements lazy evaluation.
     """
 
-    def __init__(self, params_names: List[str]):
+    def __init__(self, params_names: Iterable[str]):
         """Construct an instance from a list of strings."""
```
Comment (marcpaterno) [inline firecrown/parameters.py:67]: Comment should either say "from an Iterable yielding strings", or maybe just not describe the type (it is in the type annotations of the signature).

**Review by marcpaterno — COMMENTED** [review]
There should be two (minor) comments, one on each of two files.

**Comment by marcpaterno** [comment]
This is a comment added *after* pushing the "review" button.

**Review by marcpaterno — CHANGES_REQUESTED** [review]
This is a test of the "request changes" choice for the "Finish your review" control.

File: tests/likelihood/test_likelihood.py [hunk 2]
```diff
@@ -1,13 +1,13 @@
-import pytest
-import os 
+import os
+
 
 from firecrown.likelihood.likelihood import load_likelihood
 
 
 def test_load_likelihood_submodule():
     """The likelihood script shoult be able to load other modules from its
```
[the remark this answers is not in the document]
Comment (vitenti) [reply tests/likelihood/test_likelihood.py]: Fixed in the last commits.

File: firecrown/parameters.py (hunk 1 above)
Comment (vitenti) [reply firecrown/parameters.py]: Fixed in the last commit.

**Pull request merged**